File: BadRun.cpp
namespace isbadrun_nm{
  std::set<int> badruns;
  int lastrun=-1;
  bool lastresult=true;
  bool firstrun=true;
}

bool isbadrun(int run){

  using namespace::isbadrun_nm;

  if(firstrun){
    ifstream in("badruns.dat");
    int irun;
    std::string str;
    while(in>>irun){
      if(irun>0) badruns.insert(irun);
      getline(in,str);
    }
    std::cout<<"Loaded list of "<<badruns.size()<<" bad runs"<<std::endl;
    firstrun=false;
  }
  if(run==lastrun) return lastresult;
  lastrun=run;
  lastresult=false;
  if(badruns.find(run)!=badruns.end()) lastresult=true;

  return lastresult;
}

File: BadRun.py
class IsBadRun:
    def __init__(self):
        self.badruns = set()
        self.lastrun = -1
        self.lastresult = True
        self.firstrun = True

    def is_bad_run(self, run):
        if self.firstrun:
            with open("badruns_mine.dat", "r") as file:
                for line in file:
                    irun = int(line.strip())
                    if irun > 0:
                        self.badruns.add(irun)
                print("Loaded list of", len(self.badruns), "bad runs")
            self.firstrun = False

        if run == self.lastrun:
            return self.lastresult

        self.lastrun = run
        self.lastresult = False

        if run in self.badruns:
            self.lastresult = True

        return self.lastresult